Added new option --findip (-i) to do find IP address of each subdomain

Bogka opened the pull request:
New option will allow resolving IP addresses of found subdomains.

Option works for terminal and file output. 

In case if subdomain can't be resolved default value 0.0.0.0 will be printed. This can be used to find subdomains that are no longer used.

Example output:
```
./sublist3r.py -d example.org -i

                 ____        _     _ _     _   _____
                / ___| _   _| |__ | (_)___| |_|___ / _ __
                \___ \| | | | '_ \| | / __| __| |_ \| '__|
                 ___) | |_| | |_) | | \__ \ |_ ___) | |
                |____/ \__,_|_.__/|_|_|___/\__|____/|_|

                # Coded By Ahmed Aboul-Ela - @aboul3la

[-] Enumerating subdomains now for example.org
[-] Searching now in Baidu..
[-] Searching now in Yahoo..
[-] Searching now in Google..
[-] Searching now in Bing..
[-] Searching now in Ask..
[-] Searching now in Netcraft..
[-] Searching now in DNSdumpster..
[-] Searching now in Virustotal..
[-] Searching now in ThreatCrowd..
[-] Searching now in SSL Certificates..
[-] Searching now in PassiveDNS..
[-] Total Unique Subdomains Found: 18
www.example.org (93.184.216.34)
176-56-231-229.example.org (81.200.64.50)
anonymous.example.org (81.200.64.50)
bank.example.org (81.200.64.50)
bisertokareva.example.org (81.200.64.50)
db.example.org (81.200.64.50)
host2.example.org (81.200.64.50)
isatap.example.org (0.0.0.0)
l.example.org (81.200.64.50)
ndlt1.example.org (81.200.64.50)
m11y.ptr.record1.example.org (81.200.64.50)
test1.example.org (81.200.64.50)
testme123.example.org (81.200.64.50)
testme456.example.org (81.200.64.50)
tgtm.example.org (81.200.64.50)
versus.example.org (81.200.64.50)
wpad.example.org (0.0.0.0)
ztwuzxujmd.example.org (81.200.64.50)
```

jhaddix commented:
@Bogka Love this idea as it also helps to identify domains that are virtual hosts and the like, but it is incompatible with other options which make it less usefull, for example, cannot be combined with the port scanning flag:

``` python sublist3r.py -d TARGET.com -i -p 80 ```

will only return ports.

Bogka commented:
**Added support for port scan & IP resolution option**

Fix for - https://github.com/aboul3la/Sublist3r/pull/107#issuecomment-328302235.
_Could not test brutforce option as Subbrute code is out of date and currently not working. The new issue should be created to fix it._

Example output:
```
./sublist3r.py -d example.org -i -p 443

                 ____        _     _ _     _   _____
                / ___| _   _| |__ | (_)___| |_|___ / _ __
                \___ \| | | | '_ \| | / __| __| |_ \| '__|
                 ___) | |_| | |_) | | \__ \ |_ ___) | |
                |____/ \__,_|_.__/|_|_|___/\__|____/|_|

                # Coded By Ahmed Aboul-Ela - @aboul3la

[-] Enumerating subdomains now for example.org
[-] Searching now in Baidu..
[-] Searching now in Yahoo..
[-] Searching now in Google..
[-] Searching now in Bing..
[-] Searching now in Ask..
[-] Searching now in Netcraft..
[-] Searching now in DNSdumpster..
[-] Searching now in Virustotal..
[-] Searching now in ThreatCrowd..
[-] Searching now in SSL Certificates..
[-] Searching now in PassiveDNS..
[-] Total Unique Subdomains Found: 5
[-] Start port scan now for the following ports: 443
www.example.org (93.184.216.34) - Found open ports: 443
isatap.example.org (0.0.0.0)
ztwuzxujmd.example.org (0.0.0.0)
wpad.example.org (0.0.0.0)
bank.example.org (0.0.0.0)
```

**Fixed output of port scanner**

Previously port scanning results were only printed to console and not saved to the output file.
Added check if output file already exists. The user will have choice to overwrite the current output file or supply another file name.
Added write lock to PortScanner to prevent incorrect formatting of results.

Example output:
```
./sublist3r.py -d example.org -i -p 443 -o output.txt

                 ____        _     _ _     _   _____
                / ___| _   _| |__ | (_)___| |_|___ / _ __
                \___ \| | | | '_ \| | / __| __| |_ \| '__|
                 ___) | |_| | |_) | | \__ \ |_ ___) | |
                |____/ \__,_|_.__/|_|_|___/\__|____/|_|

                # Coded By Ahmed Aboul-Ela - @aboul3la

[-] Enumerating subdomains now for example.org
[-] File exists! Do you want to overwrite file output.txt? [y/n] y
[-] Saving results to file: output.txt
[-] Searching now in Baidu..
[-] Searching now in Yahoo..
[-] Searching now in Google..
[-] Searching now in Bing..
[-] Searching now in Ask..
[-] Searching now in Netcraft..
[-] Searching now in DNSdumpster..
[-] Searching now in Virustotal..
[-] Searching now in ThreatCrowd..
[-] Searching now in SSL Certificates..
[-] Searching now in PassiveDNS..
[-] Total Unique Subdomains Found: 5
[-] Start port scan now for the following ports: 443
isatap.example.org (0.0.0.0)
bank.example.org (0.0.0.0)
ztwuzxujmd.example.org (0.0.0.0)
wpad.example.org (0.0.0.0)
www.example.org (93.184.216.34) - Found open ports: 443
```


Example output file:
```
cat output.txt
isatap.example.org (0.0.0.0)
wpad.example.org (0.0.0.0)
bank.example.org (0.0.0.0)
ztwuzxujmd.example.org (0.0.0.0)
www.example.org (93.184.216.34) - Found open ports: 443
```


**Fixed class names to follow PEP 8 style guide**

apuhcc commented:
hello , how to update to use this functionality? 

thank you
